src/core/parser/parse_uri.c

```diff
@@ -237,28 +238,33 @@ int parse_uri(char *buf, int len, struct sip_uri *uri)
 	} else                                                 \
 		goto error_bad_char
 
-#define check_host_end         \
-	case ':':                  \
-		/* found the host */   \
-		uri->host.s = s;       \
-		uri->host.len = p - s; \
-		state = URI_PORT;      \
-		s = p + 1;             \
-		break;                 \
-	case ';':                  \
-		uri->host.s = s;       \
-		uri->host.len = p - s; \
-		state = URI_PARAM;     \
-		s = p + 1;             \
-		break;                 \
-	case '?':                  \
-		uri->host.s = s;       \
-		uri->host.len = p - s; \
-		state = URI_HEADERS;   \
-		s = p + 1;             \
-		break;                 \
-	case '&':                  \
-	case '@':                  \
+#define check_host_end                             \
+	case ':':                                      \
+		/* found the host */                       \
+		if(scheme != URN_SCH || found_host == 0) { \
+			uri->host.s = s;                       \
```

Comment:
This assigns the rest of the string to host, right?

Reply:
`s`is a char*, so this only marks where the beginning of the string would be.